next/frontend/hooks/useUser.ts

```diff
@@ -1,48 +1,109 @@
-import { useCallback, useState } from 'react'
-import { useEffectOnce } from 'usehooks-ts'
-
-// eslint-disable-next-line import/extensions
-import { getUserApi, subscribeApi, unsubscribeApi } from '../api/api'
-import { Gdpr, User } from '../dtos/generalApiDto'
-import logger from '../utils/logger'
-
-export default function useUser() {
-  const [user, setUser] = useState<User | undefined>()
-
-  const init = useCallback(async () => {
-    try {
-      const loadedUser: User = await getUserApi()
-      setUser(loadedUser)
-    } catch (error: unknown) {
-      logger.error(error)
-    }
-  }, [])
-
-  useEffectOnce(() => {
-    init().catch((error_) => logger.error('Init error', error_))
+import { cityAccountApi } from '@clients/city-account'
+import {
+  GdprDataDto,
+  GdprDataDtoCategoryEnum,
+  GdprDataDtoTypeEnum,
+} from '@clients/openapi-city-account'
+import { QueryClient, useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
+
+const userQueryKey = ['user']
+
+export const prefetchUserQuery = async (
```

Comment:
docs would be welcome (I understand form usage this is something to be used serverside so that the data is ready on first clientside render ? if so, something along these lines should be described here)